src/__tests__/postPage.test.js

```diff
@@ -0,0 +1,90 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+
+import Post from "../pages/Post";
+import fetch from "../utilities/fetch";
+
+jest.mock("../utilities/fetch");
+jest.mock("react-router-dom");
+describe("Post Page", () => {
+    test("Renders error if request fails", async () => {
+
+        render(<Post/>);
+        const error = {
+            error: "invalid score"
+        }
+
+        fetch.mockRejectedValueOnce(error);
+
+        const submitButton = screen.getByRole("button");
+
+        let errorElement = screen.queryByText(error.error);
+        expect(errorElement).toBeFalsy();
+
+        fireEvent.click(submitButton);
+
+        errorElement = screen.queryByText(error.error);
+        expect(errorElement).toBeDefined();
+    });
+
+    test("Renders success if request succeeds", async () => {
+        render(<Post/>);
+        const successText = "Created Post. Navigating to post in 3 seconds";
+
+        fetch.mockResolvedValueOnce({message: "something", post: {itemID: "something"}});
+
+        const submitButton = screen.getByRole("button");
+
+        let successElement = screen.queryByText(successText);
+        expect(successElement).toBeFalsy();
+
+        fireEvent.click(submitButton);
+
+        successElement = screen.queryByText(successText);
+        expect(successElement).toBeDefined();
```

Comment:
`toBeDefined` is not great since it returns true for anything except `undefined`, so it returns true for `null`. Use `toBeInTheDocument` instead.